src/transactions/LiquidityPoolDepositOpFrame.cpp

```diff
@@ -107,9 +107,7 @@ LiquidityPoolDepositOpFrame::depositIntoNonEmptyPool(
                               ROUND_DOWN);
         if (!minAmongValid(amountPoolShares, sharesA, resA, sharesB, resB))
         {
-            // sharesA > INT64_MAX and sharesB > INT64_MAX so we always fail
-            innerResult().code(LIQUIDITY_POOL_DEPOSIT_POOL_FULL);
-            return false;
+            throw std::runtime_error("both shares calculations overflowed");
```

Comment:
Can you add a comment here explaining why this can't happen? Something along the lines of:

It is guaranteed that either `reserveA >= totalPoolShares` or `reserveB >= totalPoolShares`. Suppose that `reserveA >= totalPoolShares` (everything works analogously for `reserveB`). Then
```
sharesA  = floor(totalPoolShares * maxAmountA / reserveA)
        <= floor(reserveA * maxAmountA / reserveA)
         = maxAmountA .
```
So at least one of the calculations should not overflow.

Reply:
Done